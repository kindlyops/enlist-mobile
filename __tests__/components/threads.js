import React from 'react'
import renderer from 'react-test-renderer'
import { fromJS } from 'immutable'
import moment from 'moment'
import ThreadBox from '../../app/components/application/threadBox'

describe('Initial state', () => {
  test('basics', () => {
    let thread = fromJS({
      id: 1
    })

    let emails = fromJS([
      {
        id: 1,
        threadId: 1,
        subject: 'Test Subject',
        text: 'Test Text',
        createdAt: moment('2019-01-01T12:00:00+00:00'),
        sendFrom: {
          type: 'User',
          id: 1
        }
      }
    ])

    let users = fromJS([
      {
        id: 1,
        fullName: 'User Name'
      }
    ])

    const tree = renderer
      .create(<ThreadBox thread={thread} emails={emails} users={users} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
