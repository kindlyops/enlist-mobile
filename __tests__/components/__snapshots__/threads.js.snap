// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial state basics 1`] = `
<View
  accessible={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderBottomColor": "#EEEEEE",
      "borderBottomWidth": 1,
      "opacity": 1,
      "paddingBottom": 10,
      "paddingTop": 10,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "padding": 15,
        "paddingBottom": 7,
        "paddingTop": 3,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "backgroundColor": "#848A96",
              "borderRadius": 7.5,
              "height": 15,
              "width": 15,
            },
            Object {
              "marginRight": 5,
              "marginTop": 1,
            },
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "backgroundColor": "transparent",
              "color": "white",
              "fontSize": 8,
              "fontWeight": "600",
            }
          }
        >
          U
        </Text>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#575D67",
          }
        }
      />
    </View>
    <View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#848A96",
            "fontSize": 13,
          }
        }
      >
        Jan 1, 17:30
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "padding": 15,
        "paddingBottom": 0,
        "paddingTop": 3,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      Test Subject
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "marginTop": 5,
          },
          Object {
            "color": "#848A96",
            "fontSize": 13,
          },
        ]
      }
    >
      Test Text
    </Text>
  </View>
</View>
`;
